cmd/hdfs/cat.go

```diff
@@ -136,7 +136,7 @@ func tailLines(file *hdfs.FileReader, numLines int64) {
 			b, err = section.ReadByte()
 		}
 
-		if err != nil && err != io.EOF {
+		if err != nil && err != io.EOF && err != io.ErrUnexpectedEOF {
```

Comment:
Rather than catching this here, can we tell the SectionReader to only read at most the number of bytes in the file? I think ErrUnexpectedEOF could also happen if something goes wrong with the connection.